src/garage/envs/multi_task_env.py

```diff
@@ -0,0 +1,139 @@
+"""
+A wrapper env that handles multiple tasks from different envs.
+
+Useful while training multi-task reinforcement learning algorithms.
+"""
+
+import akro
+import numpy as np
+
+from garage.envs import GarageEnv
+
+
+class MultiTaskEnv(GarageEnv):
+    """
+    A wrapper class to handle multi-task environments.
+
+    Args:
+        task_env_cls_dict (dict(str, garage.envs.GarageEnv)):
+            A dictionary of task_name and corresponding GarageEnv pairs.
+        task_args (dict): Arguments to be passed to GarageEnv constructer.
+            Keys should be same as keys of task_env_cls_dict.
+        task_kwargs (dict): Arguments to be passed to GarageEnv constructer.
+            Keys should be same as keys of task_env_cls_dict.
+        sample_strategy (str): The sample strategy to be used when
+            "sample_tasks" is called. Acceptable values are 'all' and 'random'.
+    """
+
+    def __init__(self,
+                 task_env_cls_dict,
+                 task_args,
+                 task_kwargs,
+                 sample_strategy='all'):
+
+        self._task_envs = []
+        self._active_task = 0
+        self._valid_sample_strategies = ['all', 'random']
+
+        if sample_strategy not in self._valid_sample_strategies:
+            raise ValueError('Invalid sample strategy.')
+
+        self._sample_strategy = sample_strategy
+
+        for task, env_cls in task_env_cls_dict.items():
+            env_kwargs = task_kwargs[task]
+            env_args = task_args[task]
+            task_env = env_cls(*env_args, **env_kwargs)
+            if self._task_envs:
+                if (task_env.observation_space.shape !=
+                        self.active_env.observation_space.shape):
+                    raise ValueError(
+                        'Observation space of all tasks should be same.')
+                if (task_env.action_space.shape !=
+                        self.active_env.action_space.shape):
+                    raise ValueError(
+                        'Action space of all tasks should be same.')
+
+            self._task_envs.append(task_env)
+
+    @property
+    def task_space(self):
+        """Task Space."""
+        one_hot_ub = np.ones(self.num_tasks)
+        one_hot_lb = np.zeros(self.num_tasks)
+        return akro.Box(one_hot_lb, one_hot_ub)
+
+    @property
+    def active_task(self):
+        """Id of currently active task."""
+        return self._active_task
+
+    @property
+    def active_env(self):
+        """Environment of currently active task."""
+        return self._task_envs[self.active_task]
+
+    @property
+    def num_tasks(self):
+        """Total number of tasks."""
+        return len(self._task_envs)
+
+    @property
+    def action_space(self):
+        """Task's Action space."""
+        return self.active_env.action_space
+
+    @property
+    def observation_space(self):
+        """Observation space."""
+        return self.active_env.observation_space
```

Comment:
I think there are more things we need to do here. If the environments do not have a unified observation space, we should at least zero-pad them.

Reply:
I think Utkarsh has already specified above that all envs must have matching action/observation spaces.

It's easy to imagine wanting a feature which automatically pads/concatenates non-matching environments, but I don't think that should be done automatically here, because it could lead to some really hard-to-find bugs for the most common use case (everything matches).